src/models/schemaRegistry.ts

```diff
@@ -39,6 +46,12 @@ export abstract class SchemaRegistry
   get schemaRegistryId(): string {
     return this.id;
   }
+
+  /** Which concrete subclass should be used to model schema registries for the corresponding connection id? */
+  static getSchemaRegistryClass(connectionId: ConnectionId): SchemaRegistrySubclass {
```

Comment:
```suggestion
  static fromConnectionId(connectionId: ConnectionId): SchemaRegistrySubclass {
```
Minor rename suggestion, I think `SchemaRegistry.fromConnectionId()` reads more intuitively than `SchemaRegistry.getSchemaRegistryClass()`

Reply:
Moved to toplevel function declared right by the record.